src/main/java/com/isroot/stash/plugin/YaccConfigServlet.java

```diff
@@ -0,0 +1,180 @@
+
+        fields = new HashMap<String, String>();
+        fieldErrors = new HashMap<String, Iterable<String>>();
+    }
+
+
+    @SuppressWarnings("unchecked")
+    public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        log.debug("doGet");
+        settingsMap = (HashMap<String, Object>) pluginSettings.get(SETTINGS_MAP);
+        if (settingsMap == null) {
+            settingsMap = new HashMap<String, Object>();
+        }
+
+        validateSettings();
+        doGetContinue(req, resp);
+    }
+
+    public void validateSettings() {
+        Settings settings = repositoryHookService.createSettingsBuilder()
+                .addAll(settingsMap)
+                .build();
+        configValidator.validate(settings, new SettingsValidationErrorsImpl(fieldErrors), null);
+    }
+
+    protected void doGetContinue(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        log.debug("doGetContinue");
+        fields.clear();
+
+        for (Map.Entry<String, Object> entry : settingsMap.entrySet()) {
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            log.debug("got plugin config " + key + "=" + value + " " + value.getClass().getName());
+            if (value instanceof String) {
+                fields.put(key, (String) value);
+            }
+        }
+
+        log.debug("Config fields: " + fields);
+        log.debug("Field errors: " + fieldErrors);
+
+        resp.setContentType("text/html;charset=UTF-8");
+        try {
+            soyTemplateRenderer.render(resp.getWriter(), "com.isroot.stash.plugin.yacc:yaccHook-config-serverside", "com.atlassian.stash.repository.hook.ref.config",
+                    ImmutableMap
+                            .<String, Object>builder()
+                            .put("config", fields)
+                            .put("errors", fieldErrors)
+                            .build()
+            );
+        } catch (SoyException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof IOException) {
+                throw (IOException) cause;
+            }
+            throw new ServletException(e);
+        }
+
+    }
+
+    public void addStringFieldValue(Map<String, Object> settingsMap, HttpServletRequest req, String fieldName) {
+        String o;
+        o = req.getParameter(fieldName);
+        if (o != null && !o.isEmpty()) settingsMap.put(fieldName, o);
+    }
+
+    public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        settingsMap.clear();
+
+        for (Object key : req.getParameterMap().keySet()) {
+            String parameterName = (String) key;
+
+            // Plugin settings persister only supports map of strings
+            if (!parameterName.startsWith("errorMessage") && !parameterName.equals("submit")) {
```

Comment:
Woohoo!  I'm catching the InvalidStateException and reporting it as a field error, which now shows up on the config form both for the global and repository configurations, giving an immediate and clear error message.  I'll include this in the push as well.